recipes/medcoupling/build.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+set -ex
+
+mkdir -p build
+cd build
+
+if [[ "$mpi" == "nompi" ]]; then
+  on_mpi="OFF"
+else
+  on_mpi="ON"
+fi
+
+# remove the share cmake files
+find ${PREFIX}/share/cmake -type d -name "medfile-*" -exec rm -rf {} +
```

Comment:
these files are typically important since they include fact about how the medfile library was compiled.

Otherwise, you are just "guessing".

Can we:

* remove the matching files from the source instead?
or 
* fix things in medfile?

Reply:
Sure, changing medfile would mean making changes to https://github.com/conda-forge/libmed-feedstock. I'll spend some time tinkering with windows compilation of medcoupling. And after that I can dig into where in libmed I would need to make changes. If you got any ideas where I might start to look I would of course appreciate it :)

Reply:
It seems this is where the path is set in the libmed-part:

https://github.com/Krande/med-copy/blob/main/CMakeLists.txt#L104

Reply:
I still really want to see if we can avoid this line

Reply:
Yes, I'm taking another stab at this now.

So the reason why I ended up deleting the folder, was becuse the following file in the libmed package (`share/cmake/medfile-4.1.1/MEDFileTargets.cmake`) has the following `set_target_properties` : 

```
set_target_properties(medC PROPERTIES
  INTERFACE_LINK_LIBRARIES "/home/conda/feedstock_root/build_artifacts/libmed_1705327158179/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/libhdf5.so;/home/conda/feedstock_root/build_artifacts/libmed_1705327158179/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/libcrypto.so;/home/conda/feedstock_root/build_artifacts/libmed_1705327158179/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/libcurl.so;/home/conda/feedstock_root/build_artifacts/libmed_1705327158179/_build_env/x86_64-conda-linux-gnu/sysroot/usr/lib/libpthread.so;/home/conda/feedstock_root/build_artifacts/libmed_1705327158179/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/libsz.so;/home/conda/feedstock_root/build_artifacts/libmed_1705327158179/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/libz.so;/home/conda/feedstock_root/build_artifacts/libmed_1705327158179/_build_env/x86_64-conda-linux-gnu/sysroot/usr/lib/libdl.so;/home/conda/feedstock_root/build_artifacts/libmed_1705327158179/_build_env/x86_64-conda-linux-gnu/sysroot/usr/lib/libm.so"
)
```

So the medC paths are (I shortened the paths by replace the main dir and the placeholder stuff with ...)

```
INTERFACE_LINK_LIBRARIES 
"../_h_env_../lib/libhdf5.so;
../_h_env_../lib/libcrypto.so;
../_h_env_../lib/libcurl.so;
../_build_env/x86_64-conda-linux-gnu/sysroot/usr/lib/libpthread.so;
../_h_env_../lib/libsz.so;
../_h_env_../lib/libz.so;
../_build_env/x86_64-conda-linux-gnu/sysroot/usr/lib/libdl.so;
../_build_env/x86_64-conda-linux-gnu/sysroot/usr/lib/libm.so"
```

So the pthread-issue I was seeing before deleting the cmake file was seemingly caused by that the sysroot paths in `MEDFileTargets.cmake` were kept as absolute.

I guess I could simply delete the `MEDFileTargets.cmake` in the libmed as a post-cmake step in the libmed run.sh. 
Or maybe I can replace the absolute paths in `MEDFileTargets.cmake` with $PREFIX and $BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/usr/lib/libpthread.so as part in the libmed build script?

Any of the 2 options you would prefer? If not I can try to find other alternatives :)

Reply:
if build_env appears in any linked path that is a bug in the libmed recipe.

From what it seems like, it seems that 3 libraries are wrongly linked:

* libpthread.so
* libdl.so
* libm.so

this should be fixed in the libmed feedstock

Reply:
FYI, from the logs, it seems that these are being pulled in by cmake + hdf5. this shouldn't really happen.